src/global/tracking/TrackerParticleCollector_factory.h

```diff
@@ -0,0 +1,42 @@
+// Created by Simon Gardner
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#pragma once
+
+#include <spdlog/logger.h>
+#include <edm4eic/TrackParametersCollection.h>
+#include <extensions/jana/JChainFactoryT.h>
+#include <algorithms/tracking/TrackerHitReconstructionConfig.h>
+
+
+namespace eicrecon {
+
+    /// This factory just collects reconstructed hits edm4eic::TrackerHit from different sources
+    /// And makes a single array out of them
+    class TrackerParticleCollector_factory : public JChainFactoryT<edm4eic::TrackParameters, TrackerHitReconstructionConfig> {
+
+    public:
+        TrackerParticleCollector_factory( std::vector<std::string> default_input_tags, TrackerHitReconstructionConfig cfg ):
```

Comment:
Would be nice to have consistent naming with respect to other implementations like https://github.com/eic/EICrecon/blob/main/src/global/pid/MergeTrack_factory.cc
Any ideas how to achieve that?

Reply:
This was just a copy of 
[https://github.com/eic/EICrecon/blob/main/src/global/tracking/TrackerHitCollector_factory.cc
](https://github.com/eic/EICrecon/blob/main/src/global/tracking/TrackerHitCollector_factory.cc
)
Would it make sense to have a template collector/merge factory?

Reply:
I think, it's fine to keep it as a copy for this PR. We would want to file an issue to track possible merging of those factories.